Remotion/Data/DomainObjects.UnitTests/Queries/Configuration/QueryConfigurationTest.cs

```diff
@@ -126,98 +147,77 @@ public void GetDefaultQueryFilePath_BaseDirectory ()
 #endif
     public void GetDefaultQueryFilePath_WithRelativeSearchPath ()
     {
-      AppDomainSetup setup = AppDomain.CurrentDomain.SetupInformation;
-      setup.ApplicationBase = Path.GetPathRoot (AppContext.BaseDirectory);
-      setup.DynamicBase = Path.GetTempPath ();
-      setup.PrivateBinPath = Path.GetFullPath (AppContext.BaseDirectory).Substring (setup.ApplicationBase.Length); // make a relative path
+      IAppContextProvider providerStub = MockRepository.GenerateStub<IAppContextProvider>();
+      providerStub.Stub (_ => _.BaseDirectory).Return (Path.GetPathRoot (_tempQueriesDirectory));
+      providerStub.Stub (_ => _.RelativeSearchPath).Return (Path.GetFullPath (_tempQueriesDirectory).Substring (providerStub.BaseDirectory.Length)); // make relative path
 
-      new AppDomainRunner (setup, delegate (object[] args)
-      {
-        QueryConfiguration configuration = new QueryConfiguration ();
-        Assert.That (!File.Exists (Path.Combine (AppContext.BaseDirectory, "queries.xml")));
-        Assert.That (configuration.GetDefaultQueryFilePath (), Is.EqualTo (Path.Combine ((string) args[0], "queries.xml")));
-      }, AppContext.BaseDirectory).Run();
+      QueryConfiguration configuration = new QueryConfiguration (providerStub);
+      Assert.That (!File.Exists (Path.Combine (providerStub.BaseDirectory, "queries.xml")));
+      Assert.That (configuration.GetDefaultQueryFilePath (), Is.EqualTo (Path.Combine (_tempQueriesDirectory, "queries.xml")));
```

Comment:
Unfortunately, I could not remember clearly if we had said the assertion in line 155 (`!File.Exists`) was still necessary or not. Did I forget to write it to the review or did we conclude it's fine?

Reply:
That line can go the way of the [Dodo ](https://en.wikipedia.org/wiki/Dodo). It's only been there to document that the file actually exists where we think it exists. Maybe it was sensible in the old test code, it's definitely not in the new test code.